src/platform/platform.cpp

```diff
@@ -680,64 +680,32 @@ void DebugPrint(const char *fmt, ...) {
 #endif
 
 //-----------------------------------------------------------------------------
-// Temporary arena, on Windows.
+// Temporary arena.
 //-----------------------------------------------------------------------------
 
-#if defined(WIN32)
-
-static HANDLE TempArena = NULL;
-
-void *AllocTemporary(size_t size)
-{
-    if(!TempArena)
-        TempArena = HeapCreate(0, 0, 0);
-    void *ptr = HeapAlloc(TempArena, HEAP_ZERO_MEMORY, size);
-    ssassert(ptr != NULL, "out of memory");
-    return ptr;
-}
+struct MimallocHeap {
+    mi_heap_t *heap = NULL;
```

Comment:
I think it'd be a little cleaner if the heap was allocated here and not in `AllocTemporary`?